monitoring/harmony-monitor/blockchain-watchdog/reporting-server.go

```diff
@@ -375,12 +380,12 @@ func (m *monitor) watchShardHealth(pdServiceKey, chain string, warning, redline
 		currentSummary := any{}
 		m.lock.Lock()
 		m.cond.Wait()
-		blockHeaderSummary(m.BlockHeaderSnapshot.Nodes, false, true, previousSummary)
+		blockHeaderSummary(m.BlockHeaderSnapshot.Nodes, false, true, previousSummary, []string{})
 		m.lock.Unlock()
 		for range time.Tick(time.Duration(interval) * time.Second) {
 			m.lock.Lock()
 			m.cond.Wait()
-			blockHeaderSummary(m.BlockHeaderSnapshot.Nodes, false, true, currentSummary)
+			blockHeaderSummary(m.BlockHeaderSnapshot.Nodes, false, true, currentSummary, []string{})
```

Comment:
why caller provides slice for callee to mutate. in-out param style not needed here, let blockHeaderSummary provide the slice